auth/README.md

```diff
@@ -0,0 +1,133 @@
+## Overview
+
+This directory implements authentication and authorization for Veraison API.
+Authentication can be performed using the Basic HTTP scheme (with the `basic`
+backend), or using a Bearer token (with the `keycloak` backend). Once an API
+user is authenticated, authorization is
+[role-based](https://en.wikipedia.org/wiki/Role-based_access_control). See
+documentation for specific services for which role(s) are needed to access
+their API.
+
+
+## Configuration
+
+- `backend`: specifies which auth backend will be used by the service. The
+  valid options are:
+
+  - `passthrough`: a backend that does not perform any authentication, allowing
+    all requests.
+  - `none`: alias for `passthrough`.
+  - `basic`: Uses the Basic HTTP authentication scheme. See
+    [RFC7617](https://datatracker.ietf.org/doc/html/rfc7617) for details. This
+    is not intended for production.
+  - `keycloak`: Uses OpenID Connect protocol as implemented by the Keycloak
+    authentication server.
+
+  See below for details of how to configure individual backends.
+
+### Passthrough
+
+No additional configuration is required. `passthrough` will allow all requests.
+This is the default if `auth` configuration is not provided.
+
+### Basic
+
+- `users`: this is a mapping of user names onto their bcrypt password hashes
+  and roles. The key of the mapping is the user name, the value is a futher
+  mapping for the details with the following fields:
+
+    - `password`: the bcrypt hash of the user's password.
+    - `roles`: either a single role or a list of roles associated with the
+      user. API authrization will be performed based on the user's roles.
+
+bcrypt hashes can be generated on the command line using `mkpasswd` utility,
+e.g.:
```

Comment:
On Linux, bcrypt hashes can be generated on the command line using `mkpasswd`
utility, e.g.: